### ISSUE TITLE
ANNOUNCEMENT: Development of v1.0.0

### ISSUE THREAD

**Issue by Zefau** [iobroker-community-adapters/ioBroker.owntracks#18]
I'm currently development a new version 1.0.0 in the fork https://github.com/Zefau/ioBroker.owntracks.
The [Test request may be found in ioBroker forum](https://forum.iobroker.net/topic/3296/iobroker-owntracks/142).
_____

Refactored entire code and removed all MQTT package dependencies (to avoid / fix security issues and reduce complexity). Thus, added MQTT adapter as dependency to manage all MQTT communication (https://github.com/ioBroker/ioBroker.mqtt/pull/52). This major change comes with the following advantages:
- use both MQTT server as well as MQTT client (to use Internet MQTT server, such as CloudMQTT) functionality (this adapter subscribes to foreign states of MQTT adapter)
- user avatars available in both server and client variant
- support TLS and websockets

Furthermore, I have added more states from the Owntracks payload into ioBroker states.
Possibly additional feature may be to share the configured locations across devices so those only have to setup once.
Also addresses https://github.com/iobroker-community-adapters/ioBroker.owntracks/issues/17.

_____

Are any further requirements need to be considered or do any further feature requests exist?

Cheers,
Zefau

**Comment by Apollon77** [comment]
Why you are not joining the adapter-community then you could use the official repo :-)

**Comment by Zefau** [comment]
Had no idea that's possible. How do I do that? ;-)

**Comment by Jey-Cee** [comment]
Hi Zefau, have a look at our [readme](https://github.com/iobroker-community-adapters/info) there you will find all infos. If not come back to us here or in the community org. I will add you to the org. ~~I need your npmjs username to add you there.~~

**Comment by Zefau** [comment]
Thanks for adding me to the community. I was wondering: Am I able to publish to NPM after merging the repo?

**Comment by Zefau** [comment]
Merged into master. @Apollon77 Could you `npm publish`? Thanks

**Comment by Apollon77** [comment]
"1.0.0-beta.2" correct?

**Comment by Zefau** [comment]
Yeah, if you don't mind I would like to public test the beta-2.

**Comment by Zefau** [comment]
I published the most recent version v1.0.0-rc.1 to an own branch (https://github.com/iobroker-community-adapters/ioBroker.owntracks/tree/v1.0.0-rc.1) and restored v0.6.2 in the master branch, because I will stop the development of v1.0.0 at this stage.

I tried to remove the MQTT dependencies from this adapter and rely the adapter on the official MQTT adapter. With this all the advantages of the MQTT can be used without redundant implementation in the owntracks adapter (e.g. secure connection via SSL or running in either client or server mode, see also [feature list](https://github.com/iobroker-community-adapters/ioBroker.owntracks/tree/v1.0.0-rc.1#100-beta1-2019-05-01)).

I ended up with a lot of MQTT warnings, e.g.
```
warn: mqtt.0 (6232) Client [XXX] Message 6 deleted after 11 retries
```
or
```
warn: mqtt.0 (6232) Client [XXX] Received puback for unknown message ID: 17
```
